content/blog/2019-01-07-we-mastered-the-art-of-programming.md

```diff
@@ -0,0 +1,63 @@
+
+But computer problem seldom get stashed away once solved. They come back to haunt you and require adjustments to the code so it fits the new constraints. Great a new problem! You have to find the fastest way to hack the code. This is fun until one or two things happen: the code you have to modify comes from a *bad* programmer or something breaks every time you modify something else. It will eventually work, but it’ll take a little longer than the previous iteration. This continues, you realise that the *bad* programmer is your past self from a few month ago and you fear modifying anything since you have no faith this [*Jenga*](https://secure.img1-fg.wfcdn.com/im/93997415/resize-h800%5Ecompr-r85/4885/48852016/Jenga%25AE+giant%25u2122+premium+jeu+de+bois+franc.jpg) will hold at all.
+
+Maybe if you took a little more time to think of how future you would feel using it things would have been different. Maybe you would have cleaned the code. Maybe you would have explicated the [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) (why divide by 8.2?). Maybe some side effects would have been commented. Well this is not important for now, because the code *works*.
+
+But this did not apply to me. I could jump back in my code, read the witty comment I wrote and remember what dimension of the matrix `X` were represented by variable `i`, `j` and `k`. I make functions, so I [don't repeat myself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). I had a bunch of smell tests that let me know if I break something. No really, I mastered the art of programming.
+
+## Until one day.
+
+I was working in a [Tech Startup](https://www.xpertsea.com/) and my team was working on a shrimp detection model. We were a few guys with different backgrounds. Everyone were software engineers except for myself; I was an actuary holding the most statistical knowledge. We were pitted against one another to figure how to solve this shrimp detection problem. Plot twist. I was not the one achieving — and for reasons I was not even aware of.
+
+It started pretty well. I quickly build a script that let me grab the data from an unstructured source and convert it to a structured format. I would then run some diagnosis to find any outliers or collinearity between features. Once the data was cleaned, another script generated a regression model using a step-wise algorithm. The model would generate some predictions and save them into a file. The last script was doing some graph and some basic analysis. Pretty standard data science duty. The report was done deal and I could quickly deliver some great results to the team.
```

Comment:
built

Reply:
run diagnostics (pas de some)

Reply:
find : detect est plus approprié

Reply:
generate predictions (pas de some)

Reply:
was doing some graph and some basic analysis : pas anglais ça. outputed data visualisations and global statistics ?

Reply:
was a done deal

Reply:
deliver great results (pas de some)